unified_planning/solvers/pddl_solver.py

```diff
@@ -79,14 +85,39 @@ def solve(self, problem: 'up.model.Problem',
             w.write_domain(domanin_filename)
             w.write_problem(problem_filename)
             cmd = self._get_cmd(domanin_filename, problem_filename, plan_filename)
-            try:
-                res = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=timeout)
-                logs.append(up.solvers.results.LogMessage(up.solvers.results.INFO, res.stdout.decode()))
-                logs.append(up.solvers.results.LogMessage(up.solvers.results.ERROR, res.stderr.decode()))
-            except subprocess.TimeoutExpired:
-                return PlanGenerationResult(up.solvers.results.TIMEOUT, plan, planner_name=self.name())
+
+            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            timer = Timer(timeout, proc.kill) # type: ignore #NOTE Timer doc does not say it can accept a None timeout, but if None is given no problems are raised.
+            timer.start()
+            proc_out: str = ''
```

Comment:
It would be better to keep a list of strings and concatenate all of them at the end